docs/decisions/0002-initial-api-specification.rst

```diff
@@ -0,0 +1,436 @@
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to view **all**
+  subsidy records, and can therefore retrieve the requested subsidy.
+
+Disallowed actions/verbs
+========================
+None of ``POST``, ``PATCH``, ``PUT``, or ``DELETE`` should be supported at this time.
+Currently, subsidy records must be created by a staff user via Django Admin. We don't support the creation
+of subsidies via a user-facing UI or from any other service.
+Neither deletions or modifications to a subsidy (e.g. the `title` field) are supported via a user-facing UI at this time.
+Any modifications that need to occur must be done by staff via Django Admin. Deletions of subsidies
+should not be enabled via the API (though perhaps are possible via Django Admin).
+
+Subsidy Transactions
+********************
+**/api/v1/subsidies/[subsidy-uuid]/transactions/**
+The root URL for reading aggregate and transaction-level data about the transactions
+associated with a subsidy.
+
+GET (list) enterprise subsidy transactions
+==========================================
+**/api/v1/subsidies/[subsidy-uuid]/transactions/?include_aggregates={true,false}**
+
+This endpoint can answer questions like:
+
+- Tell me all of the transactions for a given subsidy.
+- Within a subsidy, give me all transactions for a specific user or content key.
+- Given a user id and content key, tell me if a successful/committed transaction exists, that is,
+  has the user previously redeemed this subsidy for this content?
+
+Inputs
+------
+
+- ``subsidy-uuid`` (URL path, required): The uuid (primary key) of the subsidy for which transactions should be listed.
+- ``include_aggregates`` (query param, optional): Specifies if aggregates (quantities, number of transactions) should be
+  returned as part of the paginated response.  Defaults to ``true``.
+- ``user_id`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``user_id`` value.
+- ``content_key`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``content_key`` value.
+
+Outputs
+-------
+Returns a paginated list of aggregate and transaction data:
+
+::
+
+   {
+       'previous': null,
+       'next': '...',
+       'count': 'blah',
+       'aggregates': {
+           'total_quantity': 12350,
+           'unit': 'USD_CENTS',
+           'remaining_balance': 987650
+       },
+       'results': [
+           {
+               'uuid': 'the-transaction-uuid',
+               'status': 'completed',  TODO: enumerate valid statuses
+               'idempotency_key': 'the-idempotency-key',
+               'user_id': 54321,
+               'content_key': 'demox_1234+2T2023',
+               'quantity': 19900,
+               'unit': 'USD_CENTS',
+               'reference_id': 1234,
```

Comment:
Do piggyback on what DE/BI mentioned earlier I think that overloading these fields will make the analysis harder in the long run. I think it would be better to have sparse data than to look up data by type.

Reply:
We should have a separate ADR about "double-entry" reference tracking across services.  I'll add a ticket to our backlog.